### ISSUE TITLE
HOLD for payment 2023-11-17 - [$1000] Web -  Incorrect comment formatting after approval

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#27544]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Log in to [Expensify.com](http://expensify.com/) as user A
2. Create a policy (Settings > Policies > New Policy > Collect or Control)
3. In the policy settings go to People (Settings > Policies > "Policy name" > People tab)
4. Invite another user B
5. In the People tab of the policy ensure the admin is set as the "Submit reports to:" and the Approval Mode is "Submit & Close"
6. Create and submit a report as user B
7. Approve user B's report as user A
8. Log in to [new.expensify.com](http://new.expensify.com/) as user A
9. View the report's room (You can use the report ID in the [new.expensify.com](http://new.expensify.com/) ULR if needed e.g. [new.expensify.com/r/1234568910](http://new.expensify.com/r/1234568910))
10. Check Approval comment

## Expected Result:
Correct comment formatting after approval

## Actual Result:
 Incorrect comment formatting after approval

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.70.5
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![Screenshot 2023-09-11 at 12 26 48 PM (2)](https://github.com/Expensify/App/assets/93399543/9ec55bdb-9eeb-4884-9dd7-2dde050645b1)

**Expensify/Expensify Issue URL:**
**Issue reported by:**@joaniew
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1694460444917739

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01729ed746f5c9f54c</li>
        <li>Upwork Job ID: 1703882814739632128</li>
        <li>Last Price Increase: 2023-10-10</li>
<li>Automatic offers: </li>
<ul>
<li>jjcoffee | Reviewer | 27332302</li>
<li>Victor-Nyagudi | Contributor | 27332305</li>
</ul></li>
    </ul>
</details>

**Comment by MitchExpensify** [comment]
Haven't been able to reproduce yet https://expensify.slack.com/archives/C049HHMV9SM/p1695071461773509?thread_ts=1694460444.917739&cid=C049HHMV9SM

**Comment by MitchExpensify** [comment]
Updated the "Action Performed" to make this easier.. hopefully!

**Comment by zukilover** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Incorrect comment formatting after approval

### What is the root cause of that problem?
Text fragment is applying message header for the approval comment:
https://github.com/Expensify/App/blob/6d675529b2825b71c5c2bef332e7a07f0230c86f/src/pages/home/report/ReportActionItemFragment.js#L176-L181

### What changes do you think we should make in order to solve the problem?
Approval comment fragment comes with a defined text style. We should use that instead to style the font-weight, e.g.: 

```
                   <Text
                        numberOfLines={props.isSingleLine ? 1 : undefined}
                        style={[
                            styles.chatItemMessageHeaderSender, 
                            props.isSingleLine ? styles.pre : styles.preWrap,
                            {
                                fontWeight: props.fragment.style,
                            }
                        ]}
                    >
                        {props.fragment.text}
                    </Text>
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by zukilover** [comment]
### Proposal
[Updated](https://github.com/Expensify/App/issues/27544#issuecomment-1726669176)

**Comment by jjcoffee** [comment]
@zukilover Thanks for your [proposal](https://github.com/Expensify/App/issues/27544#issuecomment-1726669176)! I think your solution is too much of a workaround (we can't be sure what the `TEXT` fragment is based on its style) and I think the RCA is also not quite getting to the core of the issue.

Waiting on more proposals!

**Comment by MitchExpensify** [comment]
Still waiting on more proposals here

**Comment by MitchExpensify** [comment]
@zukilover do you have any ideas for a new proposal based on @jjcoffee 's feedback?

**Comment by zukilover** [comment]
@MitchExpensify I'm afraid I do not have other idea. The fragment messages are generated by the API from the OD. I regard my proposal as a backward compatible solution for this issue.

**Comment by MitchExpensify** [comment]
Doubling! 

No worries @zukilover, thanks for the context 🙇

**Comment by jjcoffee** [comment]
[Bumped](https://expensify.slack.com/archives/C01GTK53T8Q/p1696601064364559) on Slack.

**Comment by zzarcon** [comment]
# Proposal
## Please re-state the problem that we are trying to solve in this issue.
Incorrect comment formatting after approval

## What is the root cause of that problem?
Text fragment is rendered using multiple lines when it should look inline

## What changes do you think we should make in order to solve the problem?

Change `chatItemMessage` styles from `src/styles/styles.js` to row direction

```
chatItemMessage: {
  flexDirection: 'row',
} 
```

### Before

<img width="1269" alt="Screenshot 2023-10-06 at 5 38 52 pm" src="https://github.com/Expensify/App/assets/1194982/d8f9490b-ce51-4524-b157-6c8fddef1df7">


### After

<img width="1284" alt="Screenshot 2023-10-06 at 5 32 58 pm" src="https://github.com/Expensify/App/assets/1194982/68ca9110-ff17-4830-b20a-0f789cd0708c">

**Comment by jeet-dhandha** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- After approving the Report the 2nd text which should be normal styled is displaying in bold style.

### What is the root cause of that problem?

<img width="800" alt="Screenshot 2023-10-08 at 9 15 47 AM" src="https://github.com/Expensify/App/assets/78416198/f5dfad30-9258-4e54-a41e-1753149632af">


- As seen above in `/OpenReport` we have defined it to render noramlly but in `ReportActionItemFragment` in `TEXT` case we have by default given bold style and not handle `style:'normal'` case.

https://github.com/Expensify/App/blob/389d7b0c4b96c6f2a2295055c685791d35b65252/src/pages/home/report/ReportActionItemFragment.js#L155

```
chatItemMessageHeaderSender: {
    color: theme.heading,
    fontFamily: fontFamily.EXP_NEUE_BOLD,
    fontSize: variables.fontSizeNormal,
    fontWeight: fontWeightBold,
    lineHeight: variables.lineHeightXLarge,
    ...wordBreak.breakWord,
},
```

### What changes do you think we should make in order to solve the problem?
- We should add below style to `styles.js`.

```
chatItemMessageNormal: {
    color: theme.heading,
    fontFamily: fontFamily.EXP_NEUE,
    fontSize: variables.fontSizeNormal,
    lineHeight: variables.lineHeightXLarge,
    ...wordBreak.breakWord,
},
```

https://github.com/Expensify/App/blob/389d7b0c4b96c6f2a2295055c685791d35b65252/src/pages/home/report/ReportActionItemFragment.js#L155

- And update above code to below.

```
style={[
    props.fragment.style === 'normal' ? styles.chatItemMessageNormal : styles.chatItemMessageHeaderSender, 
    props.isSingleLine ? styles.pre : styles.preWrap
]}
```

<img width="800" alt="Screenshot 2023-10-08 at 9 15 28 AM" src="https://github.com/Expensify/App/assets/78416198/377ef545-8669-402a-b1cd-d23d4eb1674c">



### What alternative solutions did you explore? (Optional)
-

**Comment by Victor-Nyagudi** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Web - Incorrect comment formatting after approval

### What is the root cause of that problem?

When you approve a report on Old Dot and open it in New Dot, the text is broken up into separate lines because the `message` prop passed to `ReportActionItemMessage` is an array containing multiple message objects. 

![expensify-prop-ReportActionItemMessage-receives](https://github.com/Expensify/App/assets/79470910/1933238c-13fd-4332-9d15-0671017d997b)

Each message object's `type` is `TEXT` meaning each text item is rendered in its own `<Text>` component by `ReportActionItemFragment`.

https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/pages/home/report/ReportActionItemFragment.js#L146-L159

This is what the user sees as a result. 

![expensify-ND-broken-up-message-from-OD](https://github.com/Expensify/App/assets/79470910/d412651e-0c9d-4fab-a6ef-38ed843cb89e)

### What changes do you think we should make in order to solve the problem?


I propose 2 possible solutions. 

- Option 1

Wrap the `ReportActionItemFragment` components in [`ReportActionItemMessage`](https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/pages/home/report/ReportActionItemMessage.js#L52-L68) inside a `<Text>` component so that the text fragments align on their own as they'll now be children of a `<Text>` component, as is the natural behavior of React Native. 

```javascript
//...
<Text>
    {!props.isHidden ? (
        _.map(messages, (fragment, index) => (
            <ReportActionItemFragment
                key={`actionFragment-${props.action.reportActionID}-${index}`}
                fragment={fragment}
                iouMessage={iouMessage}
                isThreadParentMessage={ReportActionsUtils.isThreadParentMessage(props.action, props.reportID)}
                attachmentInfo={props.action.attachmentInfo}
                pendingAction={props.action.pendingAction}
                source={lodashGet(props.action, 'originalMessage.source')}
                accountID={props.action.actorAccountID}
                style={props.style}
            />
        ))
    ) : (
        <Text style={[styles.textLabelSupporting, styles.lh20]}>{props.translate('moderation.flaggedContent')}</Text>
    )}
</Text>
//...
```

Since [`ReportActionItemFragment`](https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/pages/home/report/ReportActionItemFragment.js#L88-L176) renders a `<Text>` component regardless of the message object's `type` (save for `TEXT` type that is wrapped in `UserDetailsTooltip`), everything should work as normal with a reduced likelihood of regression. 

<details>

<summary>Demo after changes</summary>

https://github.com/Expensify/App/assets/79470910/31e54127-ab41-4541-8fe1-5381398521af

</details>

- Option 2

Store the report on the back end as a regular comment i.e. type `COMMENT` so everything is rendered on one line like regular messages. I believe Old Dot isn't open source, so I wasn't able to check what goes on when a report is approved. 

### What alternative solutions did you explore? (Optional)

I considered styling the [`View`](https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/pages/home/report/ReportActionItemMessage.js#L51) in `ReportActionItemMessage` surrounding the `ReportActionItemFragment` components with the following either by reusing existing styles in `styles.js` or possibly creating a new explicit one. 

```css
flexDirection: 'row',
flexWrap: 'wrap',
```

This worked temporarily until the screen got resized, and then things started to look a bit bad. 

For example, in the following image, the space left is too much. Some of the words on the next line could easily fit there without wrapping. This is because the next `<Text>` component wraps to the next line if all its content can't fit in the allocated space. 

![expensify-text-with-flexDirection-and-flexWrap](https://github.com/Expensify/App/assets/79470910/0ec62e36-145a-409d-84fe-e315f6e951b2)

**Comment by Victor-Nyagudi** [comment]
I'd like to request some more clarity on the expected result here. 

Incorrect comment formatting is rather vague and leaves things open to interpretation. 

@jjcoffee suggested to have everything on one line in [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1696605433292189?thread_ts=1696601064.364559&cid=C01GTK53T8Q), which is what I went with when drafting my proposal. 

I'm not sure if the bolded text should also be made regular in addition to the alignment. 

Some input from the bug reporter and the design team would be much appreciated, but if that aligns with what was suggested in Slack, then all is well with me.

**Comment by jjcoffee** [comment]
@Victor-Nyagudi You have a good point that it's pretty vague. I also think it doesn't really make sense to display the `You` part of the message since it appears next to the actor's name anyway! This is also not how we display other messages:

![image](https://github.com/Expensify/App/assets/27287420/162ef627-f2f3-45ac-b1e6-a7023a9b5c60)


@MitchExpensify Do you think it's feasible to make BE changes here to return the approval message as a `COMMENT`?

**Comment by jeet-dhandha** [comment]
As reports are added from `OldDot`. I don’t think we will be able to handle it in OldDot’s BE @jjcoffee

**Comment by jeet-dhandha** [comment]
But if we have a ‘COMMENT’ type available in OldDot then it should be possible. Just that no regression should arise on OldDot’s UI due to this.

**Comment by MitchExpensify** [comment]
Friendly bump @jjcoffee

**Comment by jjcoffee** [comment]
@MitchExpensify Ah sorry I was waiting for your response on [this comment](https://github.com/Expensify/App/issues/27544#issuecomment-1753300793), or is it better if I raise that on Slack?

**Comment by jjcoffee** [comment]
We're discussing on Slack whether or not we should make BE changes.

**Comment by MitchExpensify** [comment]
"You" is expected

**Comment by Victor-Nyagudi** [comment]
## Proposal 

[Updated](https://github.com/Expensify/App/issues/27544#issuecomment-1752142100)

**Comment by Victor-Nyagudi** [comment]
I believe this issue can be addressed on the front end, @jjcoffee (see my [updated proposal](https://github.com/Expensify/App/issues/27544#issuecomment-1752142100)).

I also have some more clarity on what the correct formatting should be. Let me know if it aligns with what you think is the expected result. 

cc: @MitchExpensify

**Comment by jjcoffee** [comment]
@Victor-Nyagudi Loving the amount of detail in your [proposal](https://github.com/Expensify/App/issues/27544#issuecomment-1752142100)! Limiting using the `APPROVED` actionName is what I was looking for. Is there any reason we wouldn't want to also use that to decide whether or not to wrap `ReportActionItemFragment` in a Text component? 

I'm pretty happy with your updated proposal, I just want to get some design feedback/confirmation on how these messages should look. cc @Expensify/design

![image](https://github.com/Expensify/App/assets/27287420/b2f7d474-2d60-4b29-a169-6b92c0d10e2e)

We have a `SUBMITTED` message and an `APPROVED` message, should they both be completely greyed out as we do for other messages:

![image](https://github.com/Expensify/App/assets/27287420/9d19ddc6-5f09-4577-bc50-d60dc606d475)

Or is it valuable to keep `You` and any usernames bolded? (as suggested in the raw fragment data)?

![image](https://github.com/Expensify/App/assets/27287420/9ccd7d9e-c483-4fa8-b92b-7d9332c56db5)

**Comment by shawnborton** [comment]
yup, they should all look like the message you have with a green checkmark above. These are correct:
![image](https://github.com/Expensify/App/assets/2319350/a08fa525-8587-43c6-b5ee-e8631ce17268)

**Comment by jjcoffee** [comment]
Sweet, thanks! Let's go with @Victor-Nyagudi's [proposal](https://github.com/Expensify/App/issues/27544#issuecomment-1752142100) then.

:ribbon::eyes::ribbon: C+ reviewed

**Comment by lakchote** [comment]
@Victor-Nyagudi proposal LGTM. I like how you approached to solve this in an efficient way. Plus, you've fixed the SUBMITTED message types.

**Comment by Victor-Nyagudi** [comment]
Working on the PR right now. 

I aim to have it up and ready in roughly 2 to 4 hours.

Thank you for the assignment.

**Comment by Victor-Nyagudi** [comment]
> Limiting using the APPROVED actionName is what I was looking for. Is there any reason we wouldn't want to also use that to decide whether or not to wrap ReportActionItemFragment in a Text component?

@jjcoffee With regard to your [question]() on whether we could use the `actionName` to conditionally wrap the `ReportActionItemFrament` in a `Text` component, what do you think of this proposition?

We'll first create a variable that checks if the `props.action.actionName` is of type of `APPROVED` or `SUBMITTED` in `ReportActionItemMessage` [right below the other variables](https://github.com/Expensify/App/blob/22ccf4b0ea0badc9469942a187606accd2b97d01/src/pages/home/report/ReportActionItemMessage.js#L49).

```javascript
import CONST from '../../../CONST'; // <- Imported since it's currently not there

// ...

const isApprovedOrSubmittedReportActionType = _.contains(
    [CONST.REPORT.ACTIONS.TYPE.APPROVED, CONST.REPORT.ACTIONS.TYPE.SUBMITTED], 
    props.action.actionName
);
```

Next, we'll create a helper function in `ReportActionItemMessage` that returns a `ReportActionItemFragment`. 

This is done because `ReportActionItemFragment` will be rendered regardless of whether the `actionName` is of type `SUBMITTED`/`APPROVED` or not. Also, should changes ever need to be made, they're made in one place and apply on both sides of the `if` check. 

This function will accept two parameters - `fragment` and `index` - that it will receive from the [`map()` function in `ReportActionItemMessage`'s return statement](https://github.com/Expensify/App/blob/22ccf4b0ea0badc9469942a187606accd2b97d01/src/pages/home/report/ReportActionItemMessage.js#L53).

```javascript
// previous variables including 'isApprovedOrSubmittedReportActionType'...

const getReportActionItemFragment = (fragment, index) => {
    return(
        <ReportActionItemFragment
            key={`actionFragment-${props.action.reportActionID}-${index}`}
            fragment={fragment}
            iouMessage={iouMessage}
            isThreadParentMessage={ReportActionsUtils.isThreadParentMessage(props.action, props.reportID)}
            attachmentInfo={props.action.attachmentInfo}
            pendingAction={props.action.pendingAction}
            source={lodashGet(props.action, 'originalMessage.source')}
            accountID={props.action.actorAccountID}
            style={props.style}
            displayAsGroup={props.displayAsGroup}
            actionName={props.action.actionName}
        />
    );
};
```

We'll also need to store [the `Text` component currently rendered if `!props.isHidden` is false](https://github.com/Expensify/App/blob/22ccf4b0ea0badc9469942a187606accd2b97d01/src/pages/home/report/ReportActionItemMessage.js#L68) in a variable because that's also going to be shown regardless of the check for submitted/approved.

```javascript
// previous variables...

const flaggedContentText = (
    <Text style={[styles.textLabelSupporting, styles.lh20]}>{props.translate('moderation.flaggedContent')}</Text>
);
``` 

Finally, we'll replace [what's inside the `View` in `ReportActionItemMessage`'s return statement](https://github.com/Expensify/App/blob/22ccf4b0ea0badc9469942a187606accd2b97d01/src/pages/home/report/ReportActionItemMessage.js#L51-L70) with the following:

```javascript
<View style={[styles.chatItemMessage, !props.displayAsGroup && isAttachment ? styles.mt2 : {}, ...props.style]}>
    {
        isApprovedOrSubmittedReportActionType ? 
                
        // Wrapping 'ReportActionItemFragment' inside '<Text>' so that text isn't broken up into separate lines when  
        // there are multiple messages of type 'TEXT', as seen when a report is submitted/approved from a 
        // policy on Old Dot and then viewed on New Dot. 

        <Text>
            {!props.isHidden ? (
                _.map(messages, (fragment, index) => (getReportActionItemFragment(fragment, index)))
            ) : (
                flaggedContentText
            )}
        </Text>

        :

        <>
            {!props.isHidden ? (
                _.map(messages, (fragment, index) => (getReportActionItemFragment(fragment, index)))
            ) : (
                flaggedContentText
           )}
        </>
    }
</View>
```

Things look good so far after some light testing. 

cc: @lakchote for a second opinion

**Comment by Victor-Nyagudi** [comment]
Made some good progress on the PR, but it's late for me, so I'll pick things back up first thing tomorrow morning. 

I'm also still trying to figure out how to navigate to the Old Dot report on native devices and desktop (web works fine for mobile and Mac) given the provided method of doing so (see [step 9](https://github.com/Expensify/App/issues/27544#issue-1898743153) above) involves copying the URL from Old Dot and pasting it the New Dot URL. 

I'm open to suggestions.

**Comment by Victor-Nyagudi** [comment]
Caught this little bug currently on the latest main that would've literally rendered 'props.fragment.text' instead of the actual text. 

https://github.com/Expensify/App/blob/d28247332f58bb0d4296b0eab49527771c03464c/src/pages/home/report/ReportActionItemFragment.js#L185

I've fixed it like so in the PR. 

```javascript
<Text>{props.fragment.text}</Text>
```

**Comment by Victor-Nyagudi** [comment]
Just Android Native testing left to go, and the PR should be ready for review. Saved it for last because [it can take a while to build](https://expensify.slack.com/archives/C01GTK53T8Q/p1696657978715369?thread_ts=1696601366.090489&cid=C01GTK53T8Q).

**Comment by jjcoffee** [comment]
> If at all any of you, @jjcoffee or @lakchote, feel this change isn't needed, I can just revert it.

@Victor-Nyagudi I think you can leave this out/revert as that has been there for over 2 years (just got updated to prepend with `props`). Correct me if I'm wrong, but as I understand it this also won't affect anything in the test steps for this issue, so would be out of scope.

>I'm also still trying to figure out how to navigate to the Old Dot report on native devices and desktop

If you haven't figured this out yet, I usually use `npx uri-scheme open`, e.g. `npx uri-scheme open new-expensify://r/12345678 --android`. For the desktop app, you can just use DevTools and `window.location` in the console to open the relevant URL.

**Comment by Victor-Nyagudi** [comment]
@jjcoffee I moved those deleted comments to the draft PR. Thought it would be better to continue the conversation there, but here works too if you prefer it. 

> I think you can leave this out/revert as that has been there for over 2 years (just got updated to prepend with props)

Got it. I'll revert. 

> If you haven't figured this out yet, I usually use npx uri-scheme open, e.g. npx uri-scheme open new-expensify://r/12345678 --android. For the desktop app, you can just use DevTools and window.location in the console to open the relevant URL.

I also figured out how to get to the report on mobile devices (see [step 38](https://github.com/Expensify/App/pull/30193#issue-1957760660) and native screenshots). There are a lot of steps because of the extra set up occurring on Old Dot. 

This issue occurs on all platforms.

**Comment by Victor-Nyagudi** [comment]
I'm terribly sorry for the delay, gentlemen. 

I grossly underestimated the time it would take to test the solution on different platforms and complete the PR author checklist. 

The [PR](https://github.com/Expensify/App/pull/30193) is, however, now ready for review. 

Please take a look and let me know what you think. 

cc: @jjcoffee @lakchote

**Comment by jjcoffee** [comment]
No worries at all, I really appreciate how thorough you've been! Will be reviewing today.

**Comment by Victor-Nyagudi** [comment]
I'm glad to hear that. 

Thank you.

**Comment by jjcoffee** [comment]
Just a quick update here - the PR is still in review but we're nearing the finish line! Hopefully ready to merge by EOD.

**Comment by Victor-Nyagudi** [comment]
@OSBotify says the [PR](https://github.com/Expensify/App/pull/30325) was deployed to production 7 days ago, but @MelvinBot hasn't said anything about this.

Could something have gone wrong somewhere with the automation?

cc: @jjcoffee

**Comment by Victor-Nyagudi** [comment]
@OSBotify says the [PR](https://github.com/Expensify/App/pull/30193) was deployed to production 3 days ago, but @MelvinBot hasn't said anything about this.

Could something have gone wrong somewhere with the automation?

cc: @jjcoffee

**Comment by jjcoffee** [comment]
Oh good catch, yeah there's been some issues with good 'ole Melv lately I think. @MitchExpensify Can you update the title and label to reflect a HOLD for payment 2023-11-17? Thanks!

**Comment by MitchExpensify** [comment]
*Payment summary*:

9+ days to merge after assignment penalty (50%)

- $500 @Victor-Nyagudi  C: Upwork 
- $500 @jjcoffee  C+: Upwork

**Comment by Victor-Nyagudi** [comment]
@MitchExpensify With regard to the penalty, please take a look at this. 

I was assigned this issue on October 23rd.

<details>

<summary>Exact time I got assigned the issue</summary>

![expensify-timestamp-of-when-issue-assigned](https://github.com/Expensify/App/assets/79470910/8befc191-5f84-4043-8985-4d2c4d457ef1)

</details>

There was a [delay](https://github.com/Expensify/App/pull/30193#issuecomment-1788744635) of 2 business days at some point because the C+ was swamped with worked. 

I made the following commits on Sunday, October 29th. 

<details>

<summary>October 29th commits</summary>

![expensify-sunday-commits](https://github.com/Expensify/App/assets/79470910/448019b6-4448-4dc7-b383-00ceed9a54fc)

</details>

This wasn't a business day, so it doesn't count. However, I didn't get a response until the following Wednesday and only after I bumped the C+. 

<details>

<summary>Exact time I bumped the C+</summary>

![expensify-c-plus-bump](https://github.com/Expensify/App/assets/79470910/84be998f-9f06-4e8c-ae5d-da80e1f6eb06)

</details>

<details>

<summary>Exact time C+ responded after bump</summary>

![expensify-c-plus-review-after-bump](https://github.com/Expensify/App/assets/79470910/a40d8323-cc20-4502-94bc-24e3259e802a)

</details>

I didn't want to be a nuisance, so I wasn't bumping the C+ every day for reviews, but there wasn't much I could to prevent the 2 business days lost waiting for a response. 

[I was very much aware](https://github.com/Expensify/App/pull/30193#issuecomment-1791999727) we were approaching the 9-days penalty, so I doubled down my efforts, and thankfully, the C+ was able to prioritize the PR too. 

After everything was ironed out, the PR was ready to be merged on November 3rd. 

<details>

<summary>Exact time C+ approved PR changes</summary>

![expensify-c-plus-approves-pr](https://github.com/Expensify/App/assets/79470910/7b6d5894-efd3-4b67-ad41-6a7b54232ebe)

</details>

The amount of time in business days between October 23rd (6:45pm) when I was assigned and November 3rd (4:51pm) when the PR was ready to be merged is 8d 22h 6min.


November 3rd is a Friday, so the PR could only be merged the following week which happened. 

Again, not enough time had elapsed to warrant bumping the internal engineer, so there wasn't much I could about the extra hours incurred between PR approval and the internal engineer merging the PR.

<details>

<summary>Exact time PR got merged</summary>

![expensify-exact-time-PR-merged](https://github.com/Expensify/App/assets/79470910/63bfe5ac-5145-424d-afa4-f184bf73e5ca)

</details>

As you can see, @MitchExpensify, even without the 2-business-day delay, the PR was ready to merge in less than 9 business days. 

There's nothing I could do about the amount of work the C+ was dealing with.

In addition, if you take a look at the comments in the PR, the C+ and I were very proactive in finding a solution without regression, so the longer duration wasn't as a result of poor effort on our part. 

I beg you to take all this into account and please reconsider your stance on the 50% penalty.

**Comment by jjcoffee** [comment]
Hmm yeah I agree the penalty wouldn't usually be applied here as the PR was ready to merge within 9 WD, then was merged without changes on the next working day.

- [Assigned](https://github.com/Expensify/App/issues/27544#event-10742824948) 23/10 5:45 PM
- C+ [approved](https://github.com/Expensify/App/pull/30193#pullrequestreview-1712732593) 3/11 2:51 PM
- [Merged](https://github.com/Expensify/App/pull/30193#event-10870949437) by @lakchote without changes 6/11 11:43 AM

As @Victor-Nyagudi points out, this required a lot of work as we needed quite a lot of back and forth on the PR to handle a few edge cases so that's why we ended up going right up to the limit. Thanks for your consideration @MitchExpensify :bow:

**Comment by MitchExpensify** [comment]
Ok, that's fair! Updating the payment summary accordingly. I appreciate the timelines!

**Comment by Victor-Nyagudi** [comment]
Bump @MitchExpensify for payment.

This was supposed to be paid out last Friday.

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: **N/A - these messages were never handled**
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **N/A**
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**
- Determine if we should create a regression test for this bug. **Yes*
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. As user A: On oldDot, create a policy (Settings > Policies > New Policy > Collect or Control)
2. In the policy settings go to People (Settings > Policies > "Policy name" > People tab) and invite another user B
3. Ensure the admin is set as the "Submit reports to:" and the Approval Mode is "Submit & Approve"
4. Create and submit a report as user B to the newly created policy in step 1
5. Approve user B's report as user A
6. Log in to newDot as user A
7. Open the report using the `pageReportID` in the oldDot URL for the report 
8. Verify that the approval comment displays all on one line, greyed out like other system messages and with a consistent font weight.

Do we agree 👍 or 👎

**Comment by jjcoffee** [comment]
@MitchExpensify Checklist [complete](https://github.com/Expensify/App/issues/27544#issuecomment-1821059079)!

**Comment by MitchExpensify** [comment]
Paid and contracts ended
